homeassistant/components/homeassistant/scene.py

```diff
@@ -271,6 +282,25 @@ async def create_service(call: ServiceCall) -> None:
         SCENE_DOMAIN, SERVICE_CREATE, create_service, CREATE_SCENE_SCHEMA
     )
 
+    async def delete_service(call: ServiceCall) -> None:
+        """Delete a dynamically created scene."""
+        entity_id = call.data[CONF_ENTITY_ID]
+        scene = platform.entities.get(entity_id)
+        if scene is None:
+            _LOGGER.error("The scene %s does not exist", entity_id)
+            return
+        if not isinstance(scene, HomeAssistantScene) or not scene.from_service:
```

Comment:
I think this check should be splitted. One to check if the entity is a scene with its own log message and the other one with the actual msg.